vnext/proposals/active/keyboard-reconcile-desktop.md

```diff
@@ -0,0 +1,148 @@
+- **COMPLETED** : The event bubbling behavior with .focus() and .blur() on View in macOS has some differences from Windows and w3c standards. The behavior needs to be evaluated and reconciled between the 2 platforms.
+- The microsoft/react-native fork has implementations for all the above methods and callbacks for iOS and Android as well. Once the event bubbling behavior above has been reconciled, these can be upstreamed to core. Tracked by [Issue#243](https://github.com/microsoft/react-native-macos/issues/243)
+- TextInput also includes an `isFocused()` method that returns true/false based on whether this TextInput currently has focus. It will be useful to add this to View to and upstream as a separate PR. Tee this up with the previous one. Tracked by - [windows Issue#5511](https://github.com/microsoft/react-native-windows/issues/5511), [macOS Issue#499](https://github.com/microsoft/react-native-macos/issues/499)
+- Add `.focus()`, `.blur()`, `onFocus()`, `onBlur()` and `isFocused()` to Pressable. Tracked by [windows Issue#5589](https://github.com/microsoft/react-native-windows/issues/5589), [macOS Issue#518](https://github.com/microsoft/react-native-macos/issues/518)
+
+## Default keyboarding behaviors on JS components
+
+Both Windows and macOS implementations have added appropriate keyboarding defaults to JS components like Button, Touchables. The implementation includes firing onPress() callbacks on Button and Touchable components when they are in focus and Enter/Space key is pressed. The work remaining for these include:
+- Upstream keyboarding defaults on Button, Touchables from RNW to core. Tracked by - [windows Issue#4379](https://github.com/microsoft/react-native-windows/issues/4379)
+- Add the same support to the new Pressable component in v0.63 and upstream it so all platforms can share it. Tracked by - [windows Issue#4379](https://github.com/microsoft/react-native-windows/issues/4379)
+- macOS has some default tabbing and arrow key behaviors supported directly in VirtualizedList, FlatList, SectionList components. These need to be evaluated, completed and then upstreamed to core at which point all downstream components (including Windows) can share the behaviors. Tracked by [macOS Issue#519](https://github.com/microsoft/react-native-macos/issues/519).
+
+## onScrollKeyDown method
+
+macOS has implemented an `onScrollKeyDown` callback in ScrollView to scroll the content when arrow key is pressed. Work to reconcile this includes:
+
+- Deprecate this separate API in favor of exposing generic `onKeyDown`, `onKeyUp` methods instead (see following sections). Tracked by - [Issue#501](https://github.com/microsoft/react-native-macos/issues/501)
+- Use the generic callbacks to implement the scroll behavior in lists and scrollviews in macOS and Windows
+
+## onKeyXX callbacks
```

Comment:
This looks good to me and aligns with the W3C events (this API is what is available in React Native for Web too)

Reply:
One other thing; what about deprecating the `TextInput` prop `onKeyPress` and adding `onKeyDown`, `onKeyUp` to that component?

Reply:
`onKeyPress` on `TextInput` is likely still useful for [IME](https://en.wikipedia.org/wiki/Input_method) scenarios (which are essentially only needed for an input field).

Reply:
@harinikmsft onKeyPress is deprecated on web. How does web support IME scenarios? https://developer.mozilla.org/en-US/docs/Web/API/Document/keypress_event

Reply:
The event has an `isComposing` property set to `true` in that case. However, it's a bit more complicated (and inconsistent across browsers) than that and there has been discussion about what the timing of the composition events and when the `isComposing` property should be `true`. So in practice, on the web you also have to check whether the `keyCode` is a magic number `229` which indicates IME composition.

Reply:
Filed #5821 to track this work in Windows.